### ISSUE TITLE
Make `opaque` a soft keyword?

### ISSUE THREAD

**Issue by liufengyun** [lampepfl/dotty#5468]
[Fastparse](https://github.com/lihaoyi/fastparse/blob/a36e2cf1b82be967403a23cde3a813e5d4d3f741/fastparse/src/fastparse/package.scala#L337) has a method `opaque`, which is used in other projects, like [Scalapb](https://github.com/liufengyun/ScalaPB/blob/a561d5cd107ba4ef8c3a32f57719a2429970b692/scalapb-runtime/shared/src/main/scala/com/trueaccord/scalapb/textformat/Basics.scala#L36). After #5300 , the benchmark test for ScalaPB breaks. Can we make `opaque` a soft keyword like `inline`?

https://github.com/lihaoyi/fastparse/blob/a36e2cf1b82be967403a23cde3a813e5d4d3f741/fastparse/src/fastparse/package.scala#L337

https://github.com/liufengyun/ScalaPB/blob/a561d5cd107ba4ef8c3a32f57719a2429970b692/scalapb-runtime/shared/src/main/scala/com/trueaccord/scalapb/textformat/Basics.scala#L36

**Comment by liufengyun** [comment]
@allanrenucci I think the example in #5145 is a corner case. The current implementation already helps a lot, it covers most (if not all) practical usage.

```Scala
class Test {
  def foo(x: Int): Int = {
    val inline = 3
    5 + inline         // ok
    inline + 5         // ok
    // inline            // error
  }
}
```

**Comment by allanrenucci** [comment]
I disagree. If it does not cover all cases, I think we should disallow it. It will be very confusing for users to have a small subset of keywords that can be used as an identifier but only if it does not start a statement.

I can imagine people using the name `opaque` in an API without knowing about this corner case.

Also, at the time we discussed #5145, @odersky said it would be fairly easy to fix. Not sure why he closed the issue. Let's bring this up in the meeting

**Comment by allanrenucci** [comment]
Fixed in #5471
